tests/flagd/pkg/integration/config.go

```diff
@@ -0,0 +1,293 @@
+	configuration *flagd.ProviderConfiguration
+	error         error
+}
+
+// ctxProviderOptionsKey is the key used to pass the provider options across tests
+type ctxProviderOptionsKey struct{}
+
+// ctxErrorAwareProviderConfigurationKey is the key used to pass the errorAwareProviderConfiguration across tests
+type ctxErrorAwareProviderConfigurationKey struct{}
+
+// setEnvVar is a function to define env vars to be used to create provider configurations
+var setEnvVar func(key, value string)
+
+// valueProvider is a struct with functions to obtain the current and expected configuration value
+type valueProvider struct {
+	currentValue  func(config *flagd.ProviderConfiguration) interface{}
+	expectedValue func(value string) interface{}
+}
+
+// ignoredOptions a list of options that are currently not supported
+var ignoredOptions = []string{
+	"deadlineMs",
+	"streamDeadlineMs",
+	"keepAliveTime",
+	"retryBackoffMs",
+	"retryBackoffMaxMs",
+	"retryGracePeriod",
+	"offlinePollIntervalMs",
+	"tls", // tls option needs to be ignored because the env var FLAGD_TLS is not processed
+}
+
+// valueGeneratorMap a map that defines the value generators for supported options
+var valueGeneratorMap = map[string]valueProvider{
+	"resolver": {
+		currentValue:  func(config *flagd.ProviderConfiguration) interface{} { return config.Resolver },
+		expectedValue: func(value string) interface{} { return flagd.ResolverType(strings.ToLower(value)) },
+	},
+	"port": {
+		currentValue:  func(config *flagd.ProviderConfiguration) interface{} { return config.Port },
+		expectedValue: func(value string) interface{} { return uint16(stringToInt(value)) },
+	},
+	"host": {
+		currentValue: func(config *flagd.ProviderConfiguration) interface{} { return config.Host },
+	},
+	"targetUri": {
+		currentValue: func(config *flagd.ProviderConfiguration) interface{} { return config.TargetUri },
+	},
+	"certPath": {
+		currentValue: func(config *flagd.ProviderConfiguration) interface{} { return config.CertificatePath },
+	},
+	"socketPath": {
+		currentValue: func(config *flagd.ProviderConfiguration) interface{} { return config.SocketPath },
+	},
+	"cache": {
+		currentValue: func(config *flagd.ProviderConfiguration) interface{} {
+			return fmt.Sprintf(
+				"%v",
+				config.CacheType,
+			)
+		},
+	},
+	"selector": {
+		currentValue: func(config *flagd.ProviderConfiguration) interface{} { return config.Selector },
+	},
+	"maxCacheSize": {
+		currentValue:  func(config *flagd.ProviderConfiguration) interface{} { return config.MaxCacheSize },
+		expectedValue: func(value string) interface{} { return stringToInt(value) },
+	},
+	"offlineFlagSourcePath": {
+		currentValue: func(config *flagd.ProviderConfiguration) interface{} {
+			return config.OfflineFlagSourcePath
+		},
+	},
+}
+
+// providerOptionGeneratorMap a map that defines the provider options generators for supported options
+var providerOptionGeneratorMap = map[string]func(value string) (flagd.ProviderOption, error){
```

Comment:
do we really need this mapping? we can access the properties of the providerConfiguration directly - the helper methods are nice and simulate a more go like way, but i think we can run this test with directly setting the properties, wdyt? we even could set those fields via reflection